### PR TITLE
fix: base off existing branch only if it exists and has an open pull request

### LINKED ISSUE #336: Loss of all-contributors.rc and empty pr when user already exists
This is the continuation of the other issues after solving #332 

The `README` issue is gone 🎉  but seems like now other issues appeared:

* When I requested to add myself for something new the bot created a pr with totally new .all-contributors (removed everyone else)

This was probably because it did not fetch the local git from origin/upstream master because the commit from where it branched of is `15 commits behind master`.
image

* After when I requested for someone else who already exists in the list it seems to have fetched the latest commit before pushing but the changelog is empty.

image

image

Please have a look here: https://github.com/hertzg/node-net-keepalive/issues/152

### PR DESCRIPTION
**Pull request by hertzg**
Fixes #336 

⚠️ WIP ⚠️ 

TODO:
- [ ] Save nock response snapshots for new pull request checks in happy paths

### CODE REVIEW

File: lib/modules/repository.js [hunk 1]
```diff
@@ -174,13 +174,20 @@ class Repository {
...
+      owner: this.owner,
+      repo: this.repo,
+      state,
+      head: `${this.owner}:${branchName}`,
+    });
+  }
+
   async getPullRequestURL({ branchName }) {
     try {
-      const results = await this.octokit.pulls.list({
```
Comment (Tyjuan1980) [inline lib/modules/repository.js:179]: Looking good

File: lib/modules/repository.js [hunk 2]
```diff
@@ -174,13 +174,20 @@ class Repository {
...
-        owner: this.owner,
```
Comment (Tyjuan1980) [inline lib/modules/repository.js:180]: Nicholas t walters

File: test/unit/setup-repository.test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+const Repository = require("../../lib/modules/repository");
+const { BranchNotFoundError } = require("../../lib/modules/errors");
```
Comment (Tyjuan1980) [inline test/unit/setup-repository.test.js:2]: Gracias

**Review by Tyjuan1980 — APPROVED** [review]

**Review by tenshiAMD — CHANGES_REQUESTED** [review]
@hertzg Hi, please rebase to `master` branch first and resolve all checks so we can proceed with this PR. Thanks!

**Comment by hertzg** [comment]
Hey @tenshiAMD, unfortunately I don't have time to work on this pr now, but feel free to finish it if it's a priority right now or let me know and I can close this.

**Comment by tenshiAMD** [comment]
@gr2m I think this is the current behavior now. Please check. Thanks.

**Comment by tenshiAMD** [comment]
@hertzg Hi. Looks like the other tests got broken. Any ideas? What else are we missing here?

**Comment by gr2m** [comment]
I'll try to have look this week but can't promise

**Comment by hertzg** [comment]
@tenshiAMD Are you sure they just boke?

There’s a TODO on top of the PR

> TODO:
>
> - [ ] Save nock response snapshots for new pull request checks in happy paths

**Comment by tenshiAMD** [comment]
> @tenshiAMD Are you sure they just broke?
> 
> There’s a TODO on top of the PR
> 
> > TODO:
> >
> > - [ ] Save nock response snapshots for new pull request checks in happy paths

@hertzg yes I already update the snapshots, you can review the recent checks in this PR

**Comment by tenshiAMD** [comment]
Closing for now. Recreated to a new PR https://github.com/all-contributors/app/pull/415.

**Pull request closed without merging**